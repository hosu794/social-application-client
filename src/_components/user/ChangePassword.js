import React from "react";

import { Formik, Field, Form, ErrorMessage } from "formik";
import * as Yup from "yup";

import { useDispatch } from "react-redux";
import { authActions } from "../../_actions";

import PropTypes from "prop-types";

function ChangePassword() {
  const dispatch = useDispatch();

  function updatePassword(password) {
    dispatch(authActions.updatePassword(password));
  }

  return (
    <div className="column">
      <Formik
        className="hero-body"
        initialValues={{
          password: "",
          repeatPassword: "",
        }}
        validationSchema={Yup.object().shape({
          password: Yup.string()
            .min(6, "Username must have at least 6 characters")
            .required("Password is required"),
          repeatPassword: Yup.string()
            .min(6, "Username must have at least 6 characters")
            .oneOf([Yup.ref("password"), null], "Passwords must be same")
            .required("Password is required"),
        })}
        onSubmit={(fields) => {
          alert("SUCCESS!! :-)\n\n" + JSON.stringify(fields, null, 4));
          updatePassword(fields.password);
        }}
        render={({ errors, status, touched }) => (
          <Form>
            <h1 class="title">Change Password</h1>
            <div className="form-group">
              <label htmlFor="password">Password</label>
              <Field
                name="password"
                type="password"
                className={
                  "input" +
                  (errors.password && touched.password ? " is-danger" : "")
                }
              />
              <ErrorMessage
                name="password"
                component="div"
                className="help is-danger"
              />
            </div>
            <div className="form-group">
              <label htmlFor="repeatPassword">Reapet Username</label>
              <Field
                name="repeatPassword"
                type="password"
                className={
                  "input" +
                  (errors.repeatPassword && touched.repeatPassword
                    ? " is-danger"
                    : "")
                }
              />
              <ErrorMessage
                name="repeatPassword"
                component="div"
                className="help is-danger"
              />
            </div>

            <div className="form-group">
              <button
                type="submit"
                className="button is-info is-medium is-full-width"
                style={{
                  margin: "2rem 0 2rem 0",
                }}
              >
                Submit
              </button>
            </div>
          </Form>
        )}
      />
    </div>
  );
}

export default ChangePassword;

ChangePassword.propTypes = {
  authActions: PropTypes.object,
};
