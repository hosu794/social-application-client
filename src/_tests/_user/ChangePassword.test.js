import React from "react";

import configureStore from "redux-mock-store";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";

import ChangePassword from "../../_components/user/ChangePassword";
import { mount } from "enzyme";

let mockStore = configureStore([]);

let store = mockStore({
  authentication: {
    loading: false,
    error: true,
  },
});

describe("Test for the ChangePassword Component", () => {
  const TestingComponent = () => (
    <MemoryRouter>
      <Provider store={store}>
        <ChangePassword />
      </Provider>
    </MemoryRouter>
  );

  let wrapper = mount(<TestingComponent />);

  test("should render component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should component be defined", () => {
    expect(wrapper).toBeDefined();
  });

  test("should update a password input when it is changed ", () => {
    wrapper
      .find("input")
      .at(0)
      .simulate("change", {
        persist: () => {},
        target: {
          name: "password",
          value: "newpassword",
        },
      });

    let element = wrapper.find(ChangePassword).find("input").at(0).props()
      .value;

    expect(element).toEqual("newpassword");
  });

  test("should update a reapetPassword input when it is changed", () => {
    wrapper
      .find("input")
      .at(1)
      .simulate("change", {
        persist: () => {},
        target: {
          name: "repeatPassword",
          value: "newpassword",
        },
      });

    let element = wrapper.find(ChangePassword).find("input").at(1).props()
      .value;

    expect(element).toEqual("newpassword");
  });
});
