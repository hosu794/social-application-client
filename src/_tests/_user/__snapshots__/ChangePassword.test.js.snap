// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the ChangePassword Component should render component correctly 1`] = `ReactWrapper {}`;
